일정 멤버 (추방, 도착, 나가기) 로직 작성 및 수정 && Schedule 앱 response 리펙토링

cafitac opened the pull request:
schedule/permissions.py
 - ADD: 일정을 등록한 사람인지 인증하는 IsOwner permission 생성

schedule/views.py
 - ADD: [ScheduleViewSet] permission_class에 Isowner 등록
 - ADD: [ScheduleViewSet] 일정 멤버 추방에 관한 기능을 작성한 expulsion action 추가
 - ADD: [ScheduleViewSet] permission_class에 IsAuthenticated 추가
 - MOD: [ScheduleViewSet] retrieve일 때만 ScheduleSerializer를 사용하고 나머지 action에는 ScheduleCreateSerializer 사용
 - ADD: [ScheduleViewSet] 도착 멤버를 추가하는 arrival action 추가
 - MOD: [ScheduleViewSet] GET 파라미터로 user_id를 받아오는 것이 아닌 유저 토큰을 이용하여 api 요청을 보낸 유저를 leave하도록 변경
 - MOD: 모든 response를 responses.py에서 받아옴

schedule/models.py
 - ADD: arrival_member field 추가

schedule/migrations/0003_---.py
 - ADD: Schedule Model 마이그레이션 파일 생성

schedule/serializers.py
 - MOD: ScheduleListSerializer -> ScheduleSerializer로 이름 변경
 - ADD: ScheduleCreateSerializer의 exclude에 arrival_member field 추가

schedule/responses.py
 - ADD: 일정에 관련된 response들을 모두 정리하여 선언

Commit:
Merge remote-tracking branch 'real-blog/master'

Commit:
Merge branch 'master' of https://github.com/macbooktroops/Schedule-api-server

Commit:
일정 멤버 추방 기능 추가

schedule/permissions.py
 - ADD: 일정을 등록한 사람인지 인증하는 IsOwner permission 생성

schedule/views.py
 - ADD: ScheduleViewSet에 permission_class에 Isowner 등록
 - ADD: 일정 멤버 추방에 관한 기능을 작성한 expulsion action 추가

Commit:
일정 도착 멤버 추가 로직 작성

schedule/models.py
 - ADD: arrival_member field 추가

schedule/migrations/0003_---.py
 - ADD: Schedule Model 마이그레이션 파일 생성

schedule/serializers.py
 - MOD: ScheduleListSerializer -> ScheduleSerializer로 이름 변경
 - ADD: ScheduleCreateSerializer의 exclude에 arrival_member field 추가

schedule/views.py
 - ADD: [ScheduleViewSet] permission_class에 IsAuthenticated 추가
 - MOD: [ScheduleViewSet] retrieve일 때만 ScheduleSerializer를 사용하고 나머지 action에는 ScheduleCreateSerializer 사용
 - ADD: [ScheduleViewSet] 도착 멤버를 추가하는 arrival action 추가

Commit:
일정 response들을 모두 파일로 관리

schedule/responses.py
 - ADD: 일정에 관련된 response들을 모두 정리하여 선언

schedule/views.py
 - MOD: 모든 response를 responses.py에서 받아옴

Commit:
ScheduleViewSet leave action 로직 수정

schedule/views.py
 - MOD: GET 파라미터로 user_id를 받아오는 것이 아닌 유저 토큰을 이용하여 api 요청을 보낸 유저를 leave하도록 변경

Commit:
Merge branch 'master' of https://github.com/macbooktroops/Schedule-api-server into schedule-feature

hanseongd commented on scheduler/schedule/views.py:
```diff
@@ -42,113 +52,147 @@ def join(self, request, pk=None):
+        except User.DoesNotExist:
+            response = responses.UserNotFoundError
+
+            return Response(response, status=status.HTTP_404_NOT_FOUND)
```
굳이 response를 따로 변수에 저장할 필요가 없어보이는데
아니면 finally를 사용해서 return을 한번만 사용한다던가

cafitac replied:
아 그러면 각 exception에서 response를 선언하고 finally에서 각 상황에 맞는 Response를 날리라는거죠??

hanseongd replied:
웅 try except else finally를 알아보고 정확하게 나눠서 하면 될거같아

cafitac replied:
넵넵 finally를 활용하는 걸 깜빡했네요.. ㅠㅠ

cafitac replied:
근데 status code가 각 response 마다 다르면 이건 어떻게 처리해야 되는거예요??

hanseongd replied:
response랑 똑같이 status를 변수에 담으면 되지 않을까

cafitac replied:
반영해서 고치고 있슴다!!

hanseongd commented on scheduler/schedule/views.py:
```diff
@@ -42,113 +52,147 @@ def join(self, request, pk=None):
+
+                return Response(response, status=status.HTTP_200_OK)
+        except Schedule.DoesNotExist:
+            response = responses.ScheduleNotFoundError
```
이 부분 또한 responce로 따로 저장하지 않고 바로 return Response()에 담는게 더 확실하게 보임

Commit:
Merge branch 'master' of https://github.com/leensungzero/Schedule-api-server

Commit:
Merge branch 'schedule-feature' of https://github.com/macbooktroops/Schedule-api-server into schedule-feature

Commit:
Schedule View Response 형식 변경

schedule/views.py
 - MOD: response에 각 오류 메시지에 대한 이벤트를 담아서 한 번에 return 함으로써 중복되는 return 코드를 제거
 - MOD: Schedule arrival action 뷰에서 arrival_member을 식벽하는 부분을 GET 파라미터가 아닌 auth token을 이용하여 식별

schedule/responses.py
 - DEL: success response는 data가 함께 return 되는 경우가 많으므로 views.py에서 처리하도록 변경
 - MOD: response에 Response함수와 status를 달아줌으로써 views.py에서는 해당 response를 받아 바로 사용 가능하도록 변경

hanseongd commented on scheduler/schedule/responses.py:
```diff
@@ -0,0 +1,62 @@
+from rest_framework.response import Response
+from rest_framework import status
+
+
```
message 내용과 status code만 받아서 Response 만들어주는 함수를 만들어서 중복 코드 줄이기

hanseongd commented on scheduler/schedule/views.py:
```diff
@@ -103,50 +101,104 @@ def leave(self, request, pk=None):
+            obj = Schedule.objects.get(pk=pk)
+            self.check_object_permissions(self.request, obj)
+
+            user = request.GET.get('id')
```
user_id 혹은 user_pk로 변수명 변경

Commit:
Schedule Exception 구조 변경

schedule/responses.py -> schedule/exceptions.py
 - MOD: APIException을 상속받아 exception 처리 뷰 구조를 다시 정의함

Commit:
Schedule View 파일 변경

schedule/views.py
 - MOD: response를 리턴하는 부분을 모두 exception을 raise 하도록 변경

Commit:
Schedule Model 파일 줄정리

schedule/models.py
 - MOD: PEP8 코딩 컨벤션에 따라 소스코드 정리

Commit:
Schedule View file final

schedule/views.py
 - MOD: Schedule expulsion action 뷰 로직 개선
 - DEL: Schedule arrival action 뷰 불필요한 로직 삭제
